cpp/src/arrow/util/decimal-benchmark.cc

```diff
@@ -44,7 +44,57 @@ static void FromString(benchmark::State& state) {  // NOLINT non-const reference
   state.SetItemsProcessed(state.iterations() * values.size());
 }
 
+static void BinaryCompareOp(benchmark::State& state) {  // NOLINT non-const reference
+  BasicDecimal128 d1(546, 123), d2(123, 456);
+  while (state.KeepRunning()) {
```

Comment:
I think you can `for (auto _ : state)`. According to the Google benchmark docs:
> In C++11 mode, a ranged-based for loop should be used in preference to the KeepRunning loop for running the benchmarks.
> The reason the ranged-for loop is faster than using KeepRunning, is because KeepRunning requires a memory load and store of the iteration count ever iteration, whereas the ranged-for variant is able to keep the iteration count in a register.

https://github.com/google/benchmark/#a-faster-keeprunning-loop